javaext/com.microsoft.java.lombok/src/main/java/com/microsoft/java/lombok/DataHandler.java

```diff
@@ -0,0 +1,112 @@
+package com.microsoft.java.lombok;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IField;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.ITypeBinding;
+import org.eclipse.jdt.core.dom.rewrite.ListRewrite;
+import org.eclipse.jdt.core.manipulation.CoreASTProvider;
+import org.eclipse.jdt.internal.corext.codemanipulation.GetterSetterUtil;
+import org.eclipse.jdt.internal.corext.dom.ASTNodes;
+import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
+import org.eclipse.jdt.ls.core.internal.codemanipulation.GenerateGetterSetterOperation.AccessorField;
+import org.eclipse.jdt.ls.core.internal.codemanipulation.GenerateGetterSetterOperation.AccessorKind;
+import org.eclipse.jdt.ls.core.internal.text.correction.SourceAssistProcessor;
+import org.eclipse.lsp4j.CodeActionParams;
+import org.eclipse.text.edits.MultiTextEdit;
+import org.eclipse.text.edits.TextEdit;
+
+import com.microsoft.java.lombok.ConstructorHandler.ConstructorKind;
+
+public class DataHandler {
+
+    public static TextEdit generateDataTextEdit(CodeActionParams params, List<String> existingAnnotations,
+            IProgressMonitor monitor) {
+        IType type = SourceAssistProcessor.getSelectionType(params, monitor);
+        if (type == null || type.getCompilationUnit() == null) {
+            return null;
+        }
+
+        TextEdit textEdit = new MultiTextEdit();
+        if (!existingAnnotations.contains(AnnotationHandler.lombokAllArgsConstructorAnnotation)) {
+            TextEdit allArgsConstructorTextEdit = ConstructorHandler.generateConstructor(params, monitor,
+                    ConstructorKind.ALL_ARGS);
+            if (allArgsConstructorTextEdit != null) {
+                textEdit.addChild(allArgsConstructorTextEdit);
+            }
+        }
+
+        if (!existingAnnotations.contains(AnnotationHandler.lombokNoArgsConstructorAnnotation)) {
+            TextEdit noArgConstructorTextEdit = ConstructorHandler.generateConstructor(params, monitor,
+                    ConstructorKind.NO_ARG);
+            if (noArgConstructorTextEdit != null) {
+                textEdit.addChild(noArgConstructorTextEdit);
+            }
+        }
+
+        TextEdit accessorsTextEdit = GetterSetterHandler.generateGetterSetter(params, monitor);
```

Comment:
You can check if existing annotations contain Getter and Setter as well.

Reply:
Getter and Setter code actions are supported in https://github.com/microsoft/vscode-lombok/pull/90/commits/913d0280eb2edd1819823ac4df7a3f01f042099b